### ISSUE TITLE
api: Post Presigned post policy should use less memory and stage the i/o.

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#2290]
```
func extractHTTPFormValues(reader *multipart.Reader) (io.Reader, map[string]string, error) {
        /// HTML Form values
        formValues := make(map[string]string)
        filePart := new(bytes.Buffer)
        var err error
        for err == nil {
                var part *multipart.Part
                part, err = reader.NextPart()
                if part != nil {
                        if part.FileName() == "" {
                                var buffer []byte
                                buffer, err = ioutil.ReadAll(part) ---> Code here could server to OOM. 
                                if err != nil {
                                        return nil, nil, err
                                }
                                formValues[http.CanonicalHeaderKey(part.FormName())] = string(buffer)
                        } else {
                                if _, err = io.Copy(filePart, part); err != nil {
                                        return nil, nil, err
                                }
                        }
                }
        }
```

**Comment by harshavardhana** [comment]
Assigned to @vadmeste

**Comment by vadmeste** [comment]
@harshavardhana, it seems to me that this part of code is useless and someone has misunderstood the S3 spec, let's discuss further

**Comment by harshavardhana** [comment]
@vadmeste after the recent patch merge is this still valid?

**Comment by vadmeste** [comment]
@harshavardhana, this is not fixed yet

**Comment by harshavardhana** [comment]
Fixed by #2317
